utils/fasm/src/fasm.cpp

```diff
@@ -47,13 +47,21 @@ void FasmWriterVisitor::visit_clb_impl(ClusterBlockId blk_id, const t_pb* clb) {
 
     int x = place_ctx.block_locs[blk_id].x;
     int y = place_ctx.block_locs[blk_id].y;
+    int z = place_ctx.block_locs[blk_id].z;
     auto &grid_loc = device_ctx.grid[x][y];
     blk_type_ = grid_loc.type;
 
     current_blk_has_prefix_ = true;
     std::string grid_prefix;
     if(grid_loc.meta != nullptr && grid_loc.meta->has("fasm_prefix")) {
-        grid_prefix = grid_loc.meta->get("fasm_prefix")->front().as_string();
+      std::string prefix_unsplit = grid_loc.meta->get("fasm_prefix")->front().as_string();
+      std::vector<std::string> fasm_prefixes = vtr::split(prefix_unsplit, " ");
+      if(fasm_prefixes.size() != static_cast<size_t>(blk_type_->capacity)) {
+        vpr_throw(VPR_ERROR_OTHER,
+                  __FILE__, __LINE__, "fasm_prefix = %s, capacity = %d",
```

Comment:
Add to the message saying that these must be have equal parts.  I'd prefer the count of the fasm_prefix string for clarity too.

Reply:
ok I was following existing pattern: https://github.com/SymbiFlow/vtr-verilog-to-routing/blob/master%2Bwip/utils/fasm/src/fasm.cpp#L116-L120